hack/generator/pkg/codegen/code_generator.go

```diff
@@ -145,6 +145,9 @@ func createAllPipelineStages(idFactory astmodel.IdentifierFactory, configuration
 		// Effects the "flatten" property of Properties:
 		pipeline.FlattenProperties(),
 
+		// Remove types which may not be needed after flattening
+		pipeline.StripUnreferencedTypeDefinitions(),
```

Comment:
Might be an idea to add a postrequisite to the flattening stage to capture the requirement for post-processing.

Reply:
Doing this resulted in:
```
E0715 10:26:59.857238    9739 gen_types.go:33] Error creating code generator: postrequisite "stripUnreferenced" of stage "flattenProperties" satisfied too early
```

I think because we have multiple copies of `stripUnreferenced` and the pre/postrequisite stuff doesn't understand that?

Reply:
Will leave this as is for now, we can investigate improving the pre/post stuff to maybe support this in the future (although there are times where you only want a single copy of a stage... starting to get complicated to express in pre/post?)